[Ritvi Agarwal] iP

ritvi12 opened the pull request:
### Pixy

> “Your mind is for having ideas, not holding them.” – David Allen ([source](https://dansilvestre.com/productivity-quotes))

PIXY frees your mind of having to remember things you need to do. It's,

- text-based
- easy to learn
- ~~FAST~~ _SUPER FAST_ to use

All you need to do is,

1. download it from [here](https://github.com/nus-cs2103-AY2425S1/ip/pull/560)
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😉 
 
And it is **FREE**!

Features:

- [x] Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)


If you are a Java programmer, you can use it to practice Java too. Here's the main method:

```java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

Commit:
Level 0: Renaming ChatBot

Commit:
Level 1: Echo

Commit:
Level-1: Echo

Commit:
Level-2: Add, List

Commit:
Level 2: Add, List

Commit:
Level-3: Mark as Done

Commit:
Level-4: ToDos, Events, Deadlines

Commit:
Java Docs and fixed tagging

Commit:
A-TextUiTesting

Commit:
A-TextUiTesting

Commit:
Level-5: Handle Errors

Commit:
Level-6: Delete

Commit:
A-TextUiTesting

Commit:
Fixed automated testing

Commit:
Level-7

ritvi12 committed:
Merge pull request #1 from ritvi12/branch-Level-7

Level-7

Commit:
Level-8

ritvi12 committed:
Merge pull request #2 from ritvi12/branch-Level-8

Level-8

Commit:
A-MoreOOP

ritvi12 committed:
Merge pull request #3 from ritvi12/A-MoreOOP

A-MoreOOP

Commit:
A-Packages

Commit:
Merge branch 'master' into add-gradle-support

Commit:
Add JUnit Tests

Commit:
A-Jar

Commit:
A-JavaDoc

ritvi12 committed:
Merge pull request #4 from ritvi12/branch-A-JavaDoc

A-JavaDoc

Commit:
A-CodingStyle

ritvi12 committed:
Merge pull request #5 from ritvi12/branch-A-CodingStandard

A-CodingStyle

Commit:
Level-9: Find

ritvi12 committed:
Merge pull request #6 from ritvi12/branch-Level-9

Level-9: Find

Commit:
A-CheckStyle: Use CheckStyle

ritvi12 committed:
Merge pull request #7 from ritvi12/A-CheckStyle

A-CheckStyle: Use CheckStyle

Ansel-Ch commented on src/main/java/pixy/tasks/Deadlines.java:
```diff
@@ -0,0 +1,55 @@
+     */
+    public Deadlines(String description, String by) {
+        super(description);
+        DateTimeFormatter inputFormatter = DateTimeFormatter.ofPattern("d/M/yyyy HHmm");
+        DateTimeFormatter displayFormatter = DateTimeFormatter.ofPattern("MMM d yyyy, h:mm a");
```
Perhaps you should make these two variables constants. 😄

yikyak02 replied:
i agree, perhaps that will make the code look cleaner

Ansel-Ch reviewed (commented):
LGTM! Keep up the good work.

yikyak02 commented on src/main/java/pixy/ui/Ui.java:
```diff
@@ -0,0 +1,128 @@
+/**
+ * Handles the user interaction with the chatbot.
+ */
+public class Ui {
```
good use of Ui class to abstract out all the "---" characters!

yikyak02 commented on src/main/java/pixy/Pixy.java:
```diff
@@ -0,0 +1,88 @@
+package pixy;
```
import orders follow the java conventions, good work:)

yikyak02 commented on src/main/java/pixy/PixyExceptions.java:
```diff
@@ -0,0 +1,15 @@
+package pixy;
```
perhaps creating a separate directory for exceptions might be better for possible code expansion in the future?

yikyak02 reviewed (commented):
LGTM, keep up the good work!

Commit:
Level-10: Adding GUI

ritvi12 committed:
Merge pull request #8 from ritvi12/branch-Level-10

Level-10: Adding GUI

Commit:
Level-10

Commit:
Add assertions to code.

The code does not have assertions to validate the correctness
of task command inputs and assumptions. This could lead to errors going
undetected, especially when dealing with invalid commands or incorrect
task numbers.

Adding the assertion helps to improve code quality and prevent hidden
issues.

Let's add assert statements in various parts of the Parser and Storage
class to ensure that critical assumptions hold true.

Using assertions will help catch potential bugs early during
development.

ritvi12 committed:
Merge pull request #10 from ritvi12/branch-A-Assertions

A-Assertions

Commit:
Check Code Quality of code.

ritvi12 committed:
Merge pull request #11 from ritvi12/branch-A-CodeQuality

A-CodeQuality

Commit:
Use Streams

ritvi12 committed:
Merge pull request #12 from ritvi12/branch-A-Streams

Use Streams

Commit:
Improve GUI

Commit:
Change profile pictures

Commit:
Add Extension C-sort

Sorted deadlines and events chronologically and list them in the order
of deadlines, events and then todos.

ritvi12 committed:
Merge pull request #13 from ritvi12/A-BCD-Extension

Add Extension C-sort

Commit:
A-BCD-Extension

Commit:
A-CodeQuality

Commit:
A-BetterGui

Commit:
A-Streams

Commit:
A-BetterGui

Commit:
A-MoreErrorHandling

Commit:
A-MoreTesting

Commit:
A-UserGuide

ritvi12 committed:
Update README.md

Commit:
Update ReadME.md

Commit:
A-MoreErrorHandling

Commit:
A-CodeQuality

Commit:
A-CodeQuality: Improve Code Quality

Commit:
A-Assertions: Use Assertions

Commit:
A-Streams: Use Streams

Commit:
A-BetterGui: Better GUI

Commit:
A-AiAssisted: Enhance using AI tools
